app/services/youtube_service.rb

```diff
@@ -1,14 +1,28 @@
 class YoutubeService
 
-  def create_videos(tutorial)
+  def create_videos(tutorial, videos = [], nextPageToken = nil)
     url = "/youtube/v3/playlistItems?playlistId=#{tutorial.playlist_id}"
-    res = get_json(url, {part: "snippet"})  
+    params = {part: "snippet", maxResults: 50 }
+
+    if nextPageToken
+      params[:pageToken] = nextPageToken
+    end
+
+    res = get_json(url, params)    
+    videos << res[:items]
+    
+    if res[:nextPageToken]
+      # using recursion 
+      create_videos(tutorial, videos, res[:nextPageToken]) 
```

Comment:
This part is cool. Im using recursion which is pretty much calling a method which has the same method within it. Kinda like a loop but it is only called it a nextPageToken is passed through. So this allows us to display more that 50 videos.